src/darwin/Framework/CHIP/MTRMetricsCollector.mm

```diff
@@ -250,14 +251,39 @@ - (MTRMetrics *)metricSnapshot:(BOOL)resetCollection
 {
     std::lock_guard lock(_lock);
 
+    NSMutableArray * keysToDelete = [NSMutableArray array];
```

Comment:
Why not rename this method to since it is not public API:

(MTRMetrics *)metricSnapshotForCommissioning:(BOOL)resetCollection;